### ISSUE TITLE
Grid-like pattern

### ISSUE THREAD

**Issue by ChristopherSalomons** [shaoyanpan/Synthetic-CT-generation-from-MRI-using-3D-transformer-based-denoising-diffusion-model#4]
Hi Shaoyan, really appreciate your work here. When trying out the model, all of my predicted images exhibit a grid-like pattern matching the inference sliding window. I'm curious if you also saw this during your own training.

![image](https://github.com/shaoyanpan/Synthetic-CT-generation-from-MRI-using-3D-transformer-based-denoising-diffusion-model/assets/44589777/d5bc2891-65ec-473b-82d4-7acb1006b487)

I've tested using 20 training images for 500 epochs, as well as 144 training images for 180 epochs, and the pattern appeared for both.

**Comment by shaoyanpan** [comment]
Here are three options for improving the grids:

1. In the Jupyter notebook, under the "Build the testing function" section, modify the code as follows:
from monai.inferers import SlidingWindowInferer
img_num = 12
overlap = 0.5
inferer = SlidingWindowInferer(img_size, img_num, overlap=overlap, mode='constant')

Change the last line to:

inferer = SlidingWindowInferer(img_size, img_num, overlap=0.75, mode='gaussian')

2. The network might not be fully optimized. Try training for more epochs. In my experience, these patterns diminish significantly with longer training.

3. The grids appear only in the x-y plane, not in the z-plane, for the MRI-CT task. You can increase the patch size in the x and y dimensions and reduce it in the z dimension.

**Comment by ChristopherSalomons** [comment]
Thanks, really appreciate the help. I am curious about the reasoning behind the patch-based approach, does using smaller patches/windows provide benefits besides faster training?

**Comment by shaoyanpan** [comment]
Actually patch-based approach is not desiable, it make your performance worse, slower training and generation. The best way is to fit the whole volume (e.g., 256x256x128) for training.

However, our GPU does not have enough memory to accpet the whole volume, so patch-based approach is necessary to let our GPU can accept the data (64x64x4) then start to train. Depends on your GPU, make the patches as large as possible while maintain some reasonable batch size. For example, if your GPU is very large to accpet 16x128x128x4 volume, you can consider to increase the patch size to 256x256x4 but the batch can reduce to 4. Still good. Then you can start to train.

**Comment by shsargordi** [comment]
Hi, Christopher

May I ask what your dataset is? Is it a public dataset? I couldn't get good results with my dataset, and I am wondering about your results on your data.

**Comment by ChristopherSalomons** [comment]
I am using the [SynthRAD2023 dataset](https://doi.org/10.1002/mp.16529), my most recent results are not as good as those reported in Shaoyan's paper, but they are reasonable (~82 MAE).

**Comment by shaoyanpan** [comment]
One more thing, don't forget to average five generation samples for each patient, which should be explained in my paper. So, for each patient, you should generate five samples then average them as the final result. They can achieve better quantative result but less realistic visual appearence.

**Comment by ChristopherSalomons** [comment]
Hi Shaoyan, thanks for the advice. My training loss seems to have flattened after many epochs. I am wondering how your learning rate and weight decay values were chosen, were they selected empirically or did you test different values using a validation set?

**Comment by shaoyanpan** [comment]
I just try different values to make the quantative result to be the best for my tasks. I think you will have different optimal learning rate for your tasks. But to be honest, the improvement should be tiny, at least should not have statistical difference even you use not optimal learning rate.
